### PR TITLE
Modify to output share to file and add a method to restore share

### PR DESCRIPTION
**Pull request by KotaTakahashi9320**
# Summary
Modify to output share to file and add a method to restore share
# Purpose
For memory reduction
# Contents
Fix get_computation_result to output file when path is specified
Add method to restore share from file
Add test
# Testing Methods Performed
`pipenv run pytest`

### CODE REVIEW

File: quickmpc/qmpc_server.py [hunk 1]
```diff
@@ -129,12 +130,18 @@ def __futures_result(
         return is_ok, response
 
     @staticmethod
-    def __stream_result(stream: Iterable) -> Dict:
+    def __stream_result(stream: Iterable, job_uuid: str, party: int,
+                        path: Union[str, None]) -> Dict:
```
Comment (mdonaka) [inline quickmpc/qmpc_server.py]: I am happier with `Optional[**]` than `Union[**, None]`.

File: quickmpc/qmpc.py [hunk 2]
```diff
@@ -179,10 +180,11 @@ def get_join_table(self, join_order: Tuple[List, List, List]) -> Dict:
             ComputationMethod.Value("COMPUTATION_METHOD_JOIN_TABLE"),
             join_order, (join_order[2], []))
 
-    def get_computation_result(self, job_id: str) -> Dict:
+    def get_computation_result(self, job_id: str,
+                               path: Union[str, None] = None) -> Dict:
         logger.info("get_computation_result request. "
                     f"[job_id]={job_id}")
```
Comment (mdonaka) [inline quickmpc/qmpc.py]: I would like to see path added to the log as well.

**Review by mdonaka — COMMENTED** [review]
@KotaTakahashi9320 
Thanks for the change! Please confirm two points.

File: quickmpc/qmpc.py (hunk 2 above)
Comment (KotaTakahashi9320) [reply quickmpc/qmpc.py]: @mdonaka 
Thanks for the review!
I fixed it with a [1c8b287](https://github.com/acompany-develop/QuickMPC-libClient-py/pull/25/commits/1c8b287a10599da0960cd7d058325384b564b3a2).

File: quickmpc/qmpc_server.py (hunk 1 above)
Comment (KotaTakahashi9320) [reply quickmpc/qmpc_server.py]: @mdonaka 
Thanks for the review!
I fixed it with a [c9ef2d4](https://github.com/acompany-develop/QuickMPC-libClient-py/pull/25/commits/c9ef2d401ac7faa8c02350dfb88bfa5f39ce4a41).

**Review by mdonaka — APPROVED** [review]

**Comment by mdonaka** [comment]
@KotaTakahashi9320 
Please rebase it!

**Comment by KotaTakahashi9320** [comment]
@mdonaka 
Thanks
I rebased it.

**Pull request merged**